frontend/src/components/core/Sidebar/SidebarNav.tsx

```diff
@@ -32,13 +37,15 @@ import {
   StyledSidebarLinkText,
   StyledSidebarNavLinkContainer,
   StyledSidebarNavSeparatorContainer,
+  StyledSidebarNavButton,
 } from "./styled-components"
 
 export interface Props {
   appPages: IAppPage[]
   collapseSidebar: () => void
   currentPageScriptHash: string
   hasSidebarElements: boolean
+  collapseNav: boolean
   hideParentScrollbar: (newValue: boolean) => void
```

Comment:
Looks like we removed the logic for overflow, mouseOver/mouseOut, etc. so just want to confirm - there aren't any cases where we will be using a scrollbar in the MPA nav right?

If we have removed the scrolling from the MPA nav, we should remove the `hideParentScrollbar` prop from `SidebarNav.tsx` and its associated tests.

Reply:
Upon reviewing with the product team, we decided we'd like to keep the scrollbar on the MPA _only_ when:
* You have too many pages;
* Nav is expanded;
* There are widgets on the sidebar.

Then and only then, the sidebar will go up to 75% of the height, and allow scrolling. See video of the intended behavior below 👇🏻

https://user-images.githubusercontent.com/103376966/223125521-dbdd97bc-d1d6-4179-a4a7-4879a9a6bde7.mp4

I've reverted this functionality on [41abaf3](https://github.com/streamlit/streamlit/pull/6050/commits/41abaf37e1cadb57982ad4a4c32d1c8faa1db516) and added back the relevant tests. Let me know what you think, and if further adjustments are needed here!